backend/backend/src/main/java/de/aaaaaaah/velcom/backend/listener/Listener.java

```diff
@@ -103,47 +104,76 @@ private void update() {
 		updateDurations.update(end - start);
 	}
 
+	public void updateRepo(RepoId repoId) throws CommitSearchException {
+		LOGGER.info("Updating repo {}", repoId.getId());
+		long start = System.currentTimeMillis();
+
+		try {
+			this.lock.lock();
+
+			repoAccess.updateRepo(repoId);
+
+			manageTrackedBranches(repoId);
+			checkForUnknownCommits(repoId);
+		} finally {
+			this.lock.unlock();
+		}
+
+		long end = System.currentTimeMillis();
+		LOGGER.debug("Updating repo {} took {} ms", repoId.getId(), (end - start));
+	}
+
 	/**
-	 * Checks for new commits on the specified repository and passes the new commits to the queue.
+	 * Remove all tracked branches that don't actually exist in our repo.
 	 *
-	 * @param repoId the id of the repository to check for
+	 * @param repo the repo whose tracked branches to update
 	 */
-	public void checkForUnknownCommits(RepoId repoId)
-		throws CommitSearchException, RepoAccessException, NoSuchRepoException {
+	private void manageTrackedBranches(RepoId repoId) {
```

Comment:
`pruneTrackedBranches`? Manage is quite broad

Reply:
You're right, I'm going to change the name later